docs/supported-tools.md

```diff
@@ -0,0 +1,271 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  name="Novoda PMD rules"
+  xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+  xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+
+  ...
+
+  <!--region File exclusions-->
+  <exclude-pattern>.*\.kt</exclude-pattern>
+  <!--endregion-->
+
+  ...
+
+</ruleset>
+```
+
+## Findbugs
+[Findbugs](http://findbugs.sourceforge.net/) is a static analysis tool that looks for potential bugs in Java code. It does not support Kotlin.
+It can be used in both pure Java, and Android Java projects. It then only makes sense to have Findbugs enabled if you have Java code in your project.
+The plugin only runs Findbugs on projects that contain the Java or the Android plugin.
+
+### Configure Findbugs
+Enabling and configuring Findbugs for a project is done through the `findbugs` closure:
+
+```gradle
+findbugs {
+    toolVersion // A string, most likely '3.0.1' — the latest Findbugs release (for a long time)
+    exclude // A fileTree, such as project.fileTree('src/test/java') to exclude Java unit tests
+    excludeFilter // A file containing the Findbugs exclusions, e.g., teamPropsFile('static-analysis/findbugs-excludes.xml')
+    includeVariants { variant -> ... } // A closure to determine which variants (for Android) to include
+}
+```
+
+You can have multiple `exclude` statements.
+
+For more informations about Findbugs rules, refer to the [official website](http://findbugs.sourceforge.net/bugDescriptions.html).
+
+### Findbugs in mixed-language projects
+If your project mixes Java and Kotlin code, you most likely want to have an exclusion in place for all `*.kt` files. You can use the `exclude`
+in the configuration closure, or you can do so by adding a suppression to the filter file:
+
+`findbugs-excludes.xml`
+```xml
+<FindBugsFilter>
+
+  ...
+
+  <Match>
+    <Source name="~.*\.kt" />
+  </Match>
+
+  ...
+
+</FindBugsFilter>
+```
+
+## Detekt
+[Detekt](https://github.com/arturbosch/detekt) is a static analysis tool that looks for potential bugs and style violations in Kotlin code. It
+does not support Java. It can be used in both pure Kotlin, and Android Kotlin projects. It then only makes sense to have Detekt enabled if you
+have Kotlin code in your project. The plugin only runs Detekt on projects that contain the Kotlin or the Kotlin-Android plugin.
+
+### IMPORTANT: setup Detekt
+Unlike the other tools, the plugin **won't automatically add Detekt** to your project. If you forget to do it, the plugin will fail the build
+with an error like:
+
+```
+The detekt plugin is configured but not applied. Please apply the plugin in your build script.
+For more information see https://github.com/arturbosch/detekt.
+```
+
+In order to use Detekt in your project, you need to:
+
+ 1. Add this statement to your root `build.gradle` project (change the version according to your needs):
```

Comment:
Actually this is not entirely true. In a simpler project where there is only 1 subproject, you can directly have this there without having anything in `root` at all.

Reply:
Well, not true, because Detekt expects to be applied to the root project, _and then_ to all relevant subprojects.

Reply:
Either works. You don't have to have it in the root. If you have it in the root, then you don't need to duplicate the version. 

My goal around this comment actually was that I think we have too much information here.

Reply:
>If you have it in the root, then you don't need to duplicate the version.
In the case you were talking about initially, there is only the root, so there is no duplication. In the "normal" case that this guide covers, there is no duplication either since the version is only written in one place, the root `build.gradle`.

>My goal around this comment actually was that I think we have too much information here.
Do you mean it's bad having the minimum instructions to make the project work? I mean, this is literally the minimum amount of steps required for the Detekt integration to work. If you meant that, as things are, adding Detekt support through SAP is a painful process that requires a lot of explanations to a new starters, I absolutely agree — I maintain that not applying the plugin ourselves makes this UX rather unsatisfactory and it is _not_ fine to leave things as they are in the long term

Reply:
Let me clarify :) What I mean is that our plugin does not care how Detekt plugin is added. We just expect it to be available. 
There are many ways to add a plugin. And can also change in the future. So that's why I thought we can leave it to the official docs. But it is not really that important.

Reply:
Ok, a bit clearer now :) Well, if your concern is just that we should not mandate _how_ the Detekt plugin is added, we may just want to rephrase the text before the list from:

```
In order to use Detekt in your project, you need to:
```

to:

```
In order to use Detekt, you need to manually add it to **all** your Kotlin projects.
You can refer to the [official documentation](https://github.com/arturbosch/detekt/#gradlegroovy)
for further details. Note that you should _not_ add the `detekt` closure to your
`build.gradle`s, unlike what the official documentation says. The `detekt` closure
in the `staticAnalysis` configuration gets applied to all Kotlin modules automatically.

In most common cases, adding Detekt to a project boils down to three simple steps:
[...]
```

It is more verbose (mostly because of the not adding `detekt` stuff) than just having what we have now, but if you refer to the official docs you _must_ point out not to add the `detekt` configuration — lest things not work as expected. That was the reason why, as I explained earlier, I didn't refer users to the official docs (which by the way make a very poor job at explaining what to do).

Reply:
More verbose but at the same time more maintainable. 
I personally like the last one you suggested in above comment.

Reply:
fadaac3f765705d44d1c12b14ec0d7f8d643688c